File: src/main/java/app/application/data/UserConfig.java
package app.application.data;

import javafx.beans.property.BooleanProperty;
import javafx.beans.property.SimpleBooleanProperty;
import javafx.beans.property.SimpleStringProperty;
import javafx.beans.property.StringProperty;
import lombok.Data;

import java.io.File;

@Data
public class UserConfig {

	private StringProperty downloadDir;

	private BooleanProperty overwriteExistingVideo;

	private BooleanProperty subFolderForPlaylists;

	public UserConfig() {
		downloadDir = new SimpleStringProperty(System.getProperty("user.home") + File.separator + "Videos");
		overwriteExistingVideo = new SimpleBooleanProperty(false);
		subFolderForPlaylists = new SimpleBooleanProperty(true);
	}
}

File: src/main/java/app/application/data/UserConfig.kt
package app.application.data

import app.application.utils.GlobalValues
import javafx.beans.property.BooleanProperty
import javafx.beans.property.SimpleBooleanProperty
import javafx.beans.property.SimpleStringProperty
import javafx.beans.property.StringProperty
import lombok.Data
import java.io.File

@Data
class UserConfig {

    val downloadDir: StringProperty
    val overwriteExistingVideo: BooleanProperty
    val subFolderForPlaylists: BooleanProperty

    init {
        downloadDir = SimpleStringProperty(System.getProperty("user.home") + File.separator + GlobalValues.VIDEO_DIRECTORY)
        overwriteExistingVideo = SimpleBooleanProperty(false)
        subFolderForPlaylists = SimpleBooleanProperty(true)
    }
}